lib/mongoid/criteria/queryable/extensions/big_decimal.rb

```diff
@@ -24,7 +24,15 @@ module ClassMethods
             # @since 1.0.0
             def evolve(object)
               __evolve__(object) do |obj|
-                obj ? obj.to_s : obj
+                if obj
+                  if obj.is_a?(::BigDecimal) && Mongoid.map_big_decimal_to_decimal128
+                    BSON::Decimal128.new(obj)
+                  elsif obj.is_a?(BSON::Decimal128) && Mongoid.map_big_decimal_to_decimal128
+                    obj
```

Comment:
I think BSON::Decimal128 should always be serialized as Decimal128 regardless of config option.

Reply:
I agree